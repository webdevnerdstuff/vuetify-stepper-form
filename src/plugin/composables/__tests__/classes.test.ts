import { describe, it, expect } from 'vitest';
import {
	useColumnClasses,
	useContainerClasses,
	useStepperContainerClasses,
} from '../classes';


describe('Classes Composable', () => {

	describe('useContainerClasses', () => {
		it('should return class object with direction horizontal', () => {
			const classes = useContainerClasses({
				direction: 'horizontal',
			});

			expect(classes).toMatchSnapshot();
		});

		it('should return class object with direction vertical', () => {
			const classes = useContainerClasses({
				direction: 'vertical',
			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useStepperContainerClasses', () => {
		it('should return class object with direction horizontal', () => {
			const classes = useStepperContainerClasses({
				direction: 'horizontal',
			});

			expect(classes).toMatchSnapshot();
		});

		it('should return class object with direction vertical', () => {
			const classes = useStepperContainerClasses({
				direction: 'vertical',
			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useColumnClasses', () => {
		it('should return column classes with fieldColumns being the higher priority', () => {
			const columnsMerged = {
				lg: 12,
				md: 12,
				sm: 10,
				xl: 9,
			};

			const fieldColumns = {
				lg: 6,
				md: 6,
			};

			const classes = useColumnClasses({
				columnsMerged,
				fieldColumns,
				propName: 'foo field',
			});

			expect(classes).toMatchSnapshot();
		});
	});
});
