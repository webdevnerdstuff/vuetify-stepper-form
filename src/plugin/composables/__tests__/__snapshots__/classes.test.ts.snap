// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Classes Composable > useColumnClasses > should return column classes with fieldColumns being the higher priority 1`] = `
{
  "v-col-12": true,
  "v-col-lg-6": true,
  "v-col-md-6": true,
  "v-col-sm-10": true,
  "v-col-xl-9": true,
  "v-cols": true,
}
`;

exports[`Classes Composable > useContainerClasses > should return class object with direction horizontal 1`] = `
{
  "d-flex flex-column justify-center align-center": true,
  "v-stepper-form": true,
  "v-stepper-form--container": true,
  "v-stepper-form--container-horizontal": true,
}
`;

exports[`Classes Composable > useContainerClasses > should return class object with direction vertical 1`] = `
{
  "d-flex flex-column justify-center align-center": false,
  "v-stepper-form": true,
  "v-stepper-form--container": true,
  "v-stepper-form--container-vertical": true,
}
`;

exports[`Classes Composable > useStepperContainerClasses > should return class object with direction horizontal 1`] = `
{
  "d-flex flex-column justify-center align-center": true,
  "v-stepper-form--container-stepper": true,
  "v-stepper-form--container-stepper-horizontal": true,
}
`;

exports[`Classes Composable > useStepperContainerClasses > should return class object with direction vertical 1`] = `
{
  "d-flex flex-column justify-center align-center": false,
  "v-stepper-form--container-stepper": true,
  "v-stepper-form--container-stepper-vertical": true,
}
`;
